builder/amazon/common/run_config.go

```diff
@@ -486,6 +503,26 @@ func (c *RunConfig) Prepare(ctx *interpolate.Context) []error {
 	// Validation
 	errs := c.Comm.Prepare(ctx)
 
+	if c.Metadata.HttpEndpoint != "enabled" &&
+		c.Metadata.HttpEndpoint != "disabled" &&
+		c.Metadata.HttpEndpoint != "" {
+		msg := fmt.Errorf("http_endpoint requires either disabled or enabled as its value")
+		errs = append(errs, msg)
+	}
+
+	if c.Metadata.HttpTokens != "optional" &&
+		c.Metadata.HttpTokens != "required" &&
+		c.Metadata.HttpTokens != "" {
+		msg := fmt.Errorf("http_tokens requires either optional or required as its value")
+		errs = append(errs, msg)
+	}
+
+	if c.Metadata.HttpPutResponseHopLimit < 0 ||
+		c.Metadata.HttpPutResponseHopLimit > 64 {
+		msg := fmt.Errorf("http_put_response_hop_limit requires a number between 1 and 64")
+		errs = append(errs, msg)
+	}
```

Comment:
This is looking good! I would change a bit like:
```suggestion
      if c.Metadata.HttpEndpoint == "" {
		c.Metadata.HttpEndpoint = "enabled"
	}
	if c.Metadata.HttpTokens == "" {
		c.Metadata.HttpTokens = "optional"
	}
	if c.Metadata.HttpPutResponseHopLimit == 0 {
		c.Metadata.HttpPutResponseHopLimit = 1
	}

	if c.Metadata.HttpEndpoint != "enabled" && c.Metadata.HttpEndpoint != "disabled" {
		msg := fmt.Errorf("http_endpoint requires either disabled or enabled as its value")
		errs = append(errs, msg)
	}

	if c.Metadata.HttpTokens != "optional" && c.Metadata.HttpTokens != "required" {
		msg := fmt.Errorf("http_tokens requires either optional or required as its value")
		errs = append(errs, msg)
	}

	if c.Metadata.HttpPutResponseHopLimit < 0 ||
		c.Metadata.HttpPutResponseHopLimit > 64 {
		msg := fmt.Errorf("http_put_response_hop_limit requires a number between 1 and 64")
		errs = append(errs, msg)
	}
```
This adds the default values and whenever something is set different from the requirements it will fail the validation.